onnxruntime/core/framework/utils.cc

```diff
@@ -633,12 +635,24 @@ void DumpNodeOutputs(OpKernelContext& context, const Node& node, const SessionSt
             if (!provider) {
               provider = cpu_execution_provider;
             }
+            std::cout << " Provider=" << provider->Type() << "\n";
```

Comment:
I think we should revisit this part, to use tensor device type instead of provider type.

Reply:
Updated to use device type. Here is example output:
-----------
Output 0 Name: 439 Shape: {1,1}
OrtMemoryInfo: [ name:Cuda id:0 mem_type:0 type:1]
-4.3369937